app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SettingsImpl.java

```diff
@@ -136,4 +125,20 @@ public void setUserAgentOverride(@Nullable String value) {
     public String getUserAgentOverride() {
         return mSessionSettings.getUserAgentOverride();
     }
+
+    public String getDefaultUserAgent(int mode) {
+        return mSessionSettings.getDefaultUserAgent(toUserAgentMode(mode)) + " Wolvic/" + BuildConfig.VERSION_NAME;
+    }
+
+    private SessionSettings.UserAgentMode toUserAgentMode(int mode) {
+        switch (mode) {
+            case WSessionSettings.USER_AGENT_MODE_MOBILE:
+                return SessionSettings.UserAgentMode.MOBILE;
+            case WSessionSettings.USER_AGENT_MODE_VR:
+                return SessionSettings.UserAgentMode.MOBILE_VR;
+            case WSessionSettings.USER_AGENT_MODE_DESKTOP:
+            default:
+                return SessionSettings.UserAgentMode.DESKTOP;
```

Comment:
Why we want to accept any mode ?

Reply:
The problem is that modes are not defined as enums but as integers (which is wrong). So the compiler does not know whether you have specified all possible values in the switch and complains